lib/matplotlib/_constrained_layout.py

```diff
@@ -245,6 +254,39 @@ def check_no_collapsed_axes(layoutgrids, fig):
     return True
 
 
+def compress_fixed_aspect(layoutgrids, fig):
+    extraw = dict()
+    extrah = dict()
+    for ax in fig.axes:
+        if hasattr(ax, 'get_subplotspec'):
+            actual = ax.get_position(original=False)
+            ax.apply_aspect()
+            sub = ax.get_subplotspec()
+            gs = sub.get_gridspec()
+            if gs not in extraw:
+                extraw[gs] = np.zeros(gs.ncols)
+                extrah[gs] = np.zeros(gs.nrows)
+            orig = ax.get_position(original=True)
+            actual = ax.get_position(original=False)
+            dw = orig.width - actual.width
+            if dw > 0:
+                extraw[gs][sub.colspan] = np.maximum(extraw[gs][sub.colspan],
+                                                     dw)
+            dh = orig.height - actual.height
+            if dh > 0:
+                extrah[gs][sub.rowspan] = np.maximum(extrah[gs][sub.rowspan],
+                                                     dh)
+
+    w = np.sum(extraw[gs]) / 2
```

Comment:
It appears that only the last `gs` is used here; is it assumed to be the same `gs` for every `ax`? If that's the case, then why do `extraw`/`extrah` need to be dictionaries?

Reply:
Good catch.  I probably had dreams of something fancier here.  I guess th right thing is to error out if gs changes...